normandy/recipes/api/views.py

```diff
@@ -88,14 +89,48 @@ def update(self, request, *args, **kwargs):
         Intercept PUT requests and have them create instead of update
         if the object does not exist.
         """
-        if request.method == 'PUT':
+        if request.method in ['PUT', 'PATCH']:
             try:
-                self.get_object()
+                recipe = self.get_object()
             except Http404:
-                return self.create(request, *args, **kwargs)
+                if request.method == 'PUT':
+                    return self.create(request, *args, **kwargs)
+            else:
+                if recipe.is_approved:
+                    recipe.disable(ignore_revision_id=True)
 
         return super().update(request, *args, **kwargs)
 
+    @detail_route(methods=['POST',], permission_classes=[permissions.IsAuthenticated,])
+    def approve(self, request, pk=None):
+        recipe = self.get_object()
+
+        if recipe.is_approved:
+            return Response({'approver': 'This recipe is already approved.'},
+                            status=status.HTTP_400_BAD_REQUEST)
+        recipe.approver = request.user
+        recipe.save()
```

Comment:
Do we need a separate permission or should we just check something like `user.is_staff()`?